main.tf

```
@@ -2236,6 +2236,15 @@ resource "aws_eks_addon" "this" {
   resolve_conflicts_on_update = try(each.value.resolve_conflicts, "OVERWRITE")
   service_account_role_arn    = try(each.value.service_account_role_arn, null)
 
+  dynamic "pod_identity_association" {
+    for_each = try(each.value.pod_identity_association, [])
```

Is a list appropriate? An addon can only be associated to a single pod identity, right? Maybe an object would be better?

```suggestion
    for_each = try(each.value.pod_identity_association, null) != null ? [each.value.pod_identity_association] : []
```

---

Hello @lorengordon ,

Thank you for your time in reviewing this PR, as well as your feedback. Let me explain why I think keeping this a list would be best for forward maintainability.

> An addon can only be associated to a single pod identity, right?

It is accurate to say today there is a 1-1 relationship between AWS add-on and IAM role (and subsequent Pod Identity association). This is due to the current add-on state where each add-on has a single service account which needs IAM access.  I verified this by reviewing both [AWS documentation on add-ons](https://docs.aws.amazon.com/eks/latest/userguide/workloads-add-ons-available-eks.html) and review of the [pre-baked pod identities under terraform-aws-modules](https://github.com/terraform-aws-modules/terraform-aws-eks-pod-identity/tree/master).

However, it is technically possible for future add-ons to have multiple IAM roles/Pod Identities associated with them. This is because the Pod Identity relationship is 1-1 between IAM role and service-account. A future hypothetical example of a 1-M add-on to Pod Identity relationship follows. Imagine an add-on which combined the various observability stacks into a single, cooperative stack, i.e. a combination of `amazon-managed-service-prometheus`, `aws-cloudwatch-observability`, otel, etc.  In this situation to maintain least privilege RBAC, the add-on would need at least 2 different IAM role -> service-account Pod Identity mappings, 1 for prometheus service-account, a separate one for observablity service-account, etc.

Now granted this is entirely hypothetical, and may end up being YAGNI.  In my opinion, since the option technically exists, I recommend to make this a list, so if the module needs to support it later we don't have to make a breaking change to the input and potential associated major release of the module.  All that being said I am not a maintainer of the project, and defer this kind of maintainability decision to y'all; I am cool either way.

Given this context above, please let me know if you wish to proceed with the interface change; I would not mind making the change and re-running tests.

---

TIL, the api doc would appear to agree with you. It accepts an array instead of an object...

https://docs.aws.amazon.com/eks/latest/APIReference/API_CreateAddon.html#API_CreateAddon_RequestSyntax

Fyi, I'm not a maintainer here, I was just interested in this same feature.